frontend/components/Layout/useSystemLevelNotifications.ts

```diff
@@ -0,0 +1,64 @@
+import { useQuery } from '@tanstack/react-query';
+import { useEffect, useState } from 'react';
+
+import { getLatestEpochDetails } from '@/graphql/queries';
+import { useElectronApi } from '@/hooks/useElectronApi';
+import { useServices } from '@/hooks/useServices';
+import { useStakingProgram } from '@/hooks/useStakingProgram';
+
+type EpochStatusNotification = {
+  lastEpoch: number;
+  isNotified: boolean;
+};
+
+/**
+ * Hook to notify the user when a new epoch is started
+ * and agent is not running.
+ */
+const useNotifyOnNewEpoch = () => {
+  const { showNotification } = useElectronApi();
+  const { isServiceNotRunning } = useServices();
+  const { activeStakingProgramAddress } = useStakingProgram();
+
+  const [epochStatusNotification, setEpochStatusNotification] =
+    useState<EpochStatusNotification | null>(null);
+
+  const { data: epoch } = useQuery({
```

Comment:
so, how this query should be updated while the app is running? wouldn't it be better to use a listener for a new checkpoint event instead?